enterprise/frontend/src/embedding-sdk/lib/polyfill-use-sync-external-store.ts

```diff
@@ -0,0 +1,10 @@
+import React from "react";
+import { useSyncExternalStore } from "use-sync-external-store/shim";
+
+// Monkey-patches useSyncExternalStore if we are in React 17,
+// where useSyncExternalStore is not available.
+// This is used in react-redux v8, until they've dropped the shim in v9.
+if (!("useSyncExternalStore" in React)) {
+  // @ts-expect-error - we're patching for React 17, which doesn't have this method.
+  React.useSyncExternalStore = useSyncExternalStore;
```

Comment:
Unfortunately this needs to be a global monkey-patch, as we can't monkey-patch a built CJS bundle in the SDK user's environment. This is less error-prone than that as well. Luckily, the `useSyncExternalStore` shim here is officially provided by the React team, so it's 100% standard.